security-admin/src/main/java/org/apache/ranger/rest/XUserREST.java

```diff
@@ -863,6 +868,49 @@ public void deleteXUserByUserName(@PathParam("userName") String userName,
 		xUserMgr.deleteXUser(vxUser.getId(), forceDelete);
 	}
 
+
+	/**
+	 * Proceed with caution: Force deletes users in bulk from the ranger db,
+	 * essentially serves as a cleanup Op for external users.
+	 * <tt>Delete</tt> happens one at a time with immediate commit on the transaction.
+	 */
+	@DELETE
+	@Path("/delete/external/users")
+	@PreAuthorize("hasRole('ROLE_SYS_ADMIN')")
+	@Produces({ "application/json" })
+	public Response deleteXUsers() {
+		if (!usersDeletionInProgress){
+			synchronized (XUserREST.class) {
+				usersDeletionInProgress = true;
+				xUserMgr.bulkDeleteUsers();
+				usersDeletionInProgress = false;
+			}
+			return Response.ok("External users deleted successfully").build();
+		}
+		return Response.ok("Users Deletion in progress, check ranger admin logs!").build();
```

Comment:
Instead of returning 'ok', consider returning an error code like 409 (conflict).